src/EditorFeatures/Core/Shared/Tagging/TagSources/ProducerPopulatedTagSource.cs

```diff
@@ -675,7 +686,7 @@ public override ITagSpanIntervalTree<TTag> GetTagIntervalTreeForBuffer(ITextBuff
 
                     // We shall synchronously compute tags.
                     var producedTags = ConvertToTagTree(
```

Comment:
this method probably need to re-written so that we can properly support GetAccurateTagIntervalTreeForBuffer. existing code seems a bit of hack to support this ComputeTagsSynchronouslyIfNoAsynchronousComputationHasCompleted when the behavior is not officially supported.